File: 3278-find-the-number-of-ways-to-place-people-i/3278-find-the-number-of-ways-to-place-people-i.cpp
class Solution {
public:
    int numberOfPairs(vector<vector<int>>& points) {
        int res = 0;
        sort(points.begin(), points.end(), [](vector<int>& a, vector<int>&b){
            if (a[0] == b[0])
            {
                return a[1] > b[1];
            }
            return a[0] < b[0];
        });
        for (char i = 0; i < points.size(); i++)
        {
            char lowerLimit = -1;
            for (char j = i + 1; j < points.size(); j++)
            {
                if (points[j][1] < points[i][1])
                {
                    if (points[j][1] > lowerLimit)
                    {
                        res++;
                        lowerLimit = points[j][1];
                    }
                }
                else if(points[j][1] == points[i][1])
                {
                    res++;
                    break;
                }
            }
        } 
        return res;
    }
};

File: 3278-find-the-number-of-ways-to-place-people-i/3278-find-the-number-of-ways-to-place-people-i.py
class Solution:
    def numberOfPairs(self, points: List[List[int]]) -> int:
        res = 0
        points = sorted(points, key=lambda point: (point[0], -point[1]))
        for i in range(len(points)):
            lowerLimit = -1
            for j in range(i + 1, len(points)):
                if points[j][1] < points[i][1]:
                    if points[j][1] > lowerLimit:
                        res += 1
                        lowerLimit = points[j][1]
                elif points[j][1] == points[i][1]:
                    res += 1
                    break
        return res
